sound/soc/sof/pcm.c

```diff
@@ -470,6 +470,10 @@ static int sof_pcm_trigger(struct snd_soc_component *component,
 		if (ret < 0)
 			return ret;
 
+		ret = snd_sof_pcm_platform_hw_free(sdev, substream);
```

Comment:
The commit message is still very hard to parse: "To enable restarting the stream without a hw_free or resuming
a suspending stream." So are the two cases:
 - restarting the stream without a hw_free, and
 - resuming a suspending stream
I read the second case originally as  "restarting the stream without resume a suspending stream".